examples/other_datasets/financial_data.py

```diff
@@ -0,0 +1,184 @@
+target = dataset.FRAUD
+
+# Let's encode our categorical variable (LabelEncoding):
+features["IP_TERMINAL"] = features["IP_TERMINAL"].astype("category").cat.codes
+
+# ... and create an 'index' column in the dataset
+# Note: this is done only for progamming reason, due to our implementation
+# of the `ToEpochs` transformer (see below)
+features["index"] = features.index
+
+##############################################################################
+# Create the pipeline
+# ----------------
+#
+# Let's create the pipeline as suggested in the patent application
+
+# Let's start by creating the required transformers:
+
+
+class ToEpochs(TransformerMixin, BaseEstimator):
+    def __init__(self, n):
+        self.n = n
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        all_epochs = []
+        for x in X:
+            id = x[3]
+            epoch = features[features.index > id - self.n]
+            epoch = epoch[epoch.index <= id]
+            epoch.drop(columns=["index"], inplace=True)
+            all_epochs.append(epoch)
+        all_epochs = np.array(all_epochs)
+        return all_epochs
+
+
+def slim(x, keep_diagonal=True):
+    # Vectorize covariance matrices by removing redundant information.
+    length = len(x) // 2
+    first = range(0, length)
+    last = range(len(x) - length, len(x))
+    down_cadrans = x[np.ix_(last, last)]
+    if keep_diagonal:
+        down_cadrans = [down_cadrans[i, j] for i in first for j in first if i <= j]
+    else:
+        down_cadrans = [down_cadrans[i, j] for i in first for j in first if i < j]
+    first_cadrans = np.reshape(x[np.ix_(last, first)], (1, len(x)))
+    ret = np.append(first_cadrans, down_cadrans)
+    return ret
+
+
+class SlimVector(TransformerMixin, BaseEstimator):
+    def __init__(self, keep_diagonal):
+        self.keep_diagonal = keep_diagonal
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        return np.array([slim(x, self.keep_diagonal) for x in X])
+
+
+class OptionalWhitening(TransformerMixin, BaseEstimator):
+    def __init__(self, process=True, n_components=4):
+        self.process = process
+        self.n_components = n_components
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        if not self.process:
+            return X
+        return Whitening(dim_red={"n_components": 4}).fit_transform(X)
+
+
+# Finally put together the transformers, and add at the end
+# the SVM classifier (classical)
+pipe = make_pipeline(
+    ToEpochs(n=10),
+    XdawnCovariances(nfilter=1),
+    OptionalWhitening(process=True, n_components=4),
+    SlimVector(keep_diagonal=True),
+    QuanticSVM(quantum=False),
+)
+
+# Optimize the pipeline.
+# Let's save some time and run the optimization with a classical SVM.
+gs = GridSearchCV(
+    pipe,
+    param_grid={
+        "toepochs__n": [10, 20],
+        "xdawncovariances__nfilter": [1, 2],
```

Comment:
You should test higher values for nfilter.
What is the number of features?

Reply:
Makes sense. I can try, and find a compromise for the pipeline afterwhat.